### ISSUE TITLE
Handling of symbol clashes in psci

### ISSUE THREAD

**Issue by hdgarrood** [purescript/purescript#756]
Related to #753. psci could handle situations where symbols clash better.

For example, `Data.Array` and `Data.Map` both export `singleton`. Currently if you attempt to import both, psci fails to import the second because of the symbol clash.

A better behaviour could be to allow such an import to succeed, and produce an error only when attempting to use a clashing symbol. This seems to be what ghci does.

So an interaction could look like:
```
> :i Data.Array
> :i Data.Map
> let f = singleton 5
Error: 'singleton' could refer to 'Data.Map.singleton' or 'Data.Array.singleton'
> Data.Map.singleton 5 7
fromList [Tuple (5) (7)]
```

**Comment by hdgarrood** [comment]
This seems to work as I described now, with the 0.8.0 RC. :)

**Comment by hdgarrood** [comment]
Probably fixed by #1688

**Comment by garyb** [comment]
Yep, that'd do it! :smile:
